cpp/tests/io/nested_json_test.cpp

```diff
@@ -450,20 +450,28 @@ TEST_F(JsonTest, TokenStream2)
   }
 }
 
-TEST_F(JsonTest, ExtractColumn)
+struct JsonParserTest : public cudf::test::BaseFixture, public testing::WithParamInterface<bool> {
+};
+INSTANTIATE_TEST_SUITE_P(Experimental, JsonParserTest, testing::Bool());
+
+TEST_P(JsonParserTest, ExtractColumn)
 {
   using cuio_json::SymbolT;
+  bool const is_full_gpu = GetParam();
+  auto json_parser =
+    is_full_gpu ? cuio_json::detail::parse_nested_json2 : cuio_json::detail::parse_nested_json;
```

Comment:
updated. Nice suggestion. 
(to be fair in commments, `host_parse_nested_json` also uses gpu tokenizer.)

Reply:
true. I hope these names work anyway, since we've been referring to the new implementation as the "device implementation". We should not need them long term anyway 🤞